services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/ValidationService.java

```diff
@@ -0,0 +1,120 @@
+
+public class ValidationService {
+    private static final Logger logger = LoggerFactory.getLogger(ValidationService.class);
+    public static final Set<Long> SERVICE_RULES =
+            ImmutableSet.of(0x8000000000000001L, 0x8000000000000002L, 0x8000000000000003L);
+    private SwitchFlowEntries request;
+    private Session session;
+    private SwitchId switchId;
+
+    public ValidationService(SwitchFlowEntries request, Session session) {
+        this.request = request;
+        this.session = session;
+        switchId = request.getSwitchId();
+    }
+
+    /**
+     * Runs validating process.
+     * @return validation result as RulesResponse
+     */
+    public SyncRulesResponse validate() {
+        logger.debug("Validating rules on switch {}", switchId);
+
+        Set<Long> expectedCookies = readSegmentsCookies();
+        expectedCookies.addAll(readIngressCookies());
+        logger.debug("Expected rules on switch {}: {}", switchId, expectedCookies);
+
+        Set<Long> presentedCookies = request.getFlowEntries().stream()
+                .map(FlowEntry::getCookie)
+                .collect(Collectors.toSet());
+        presentedCookies.removeAll(SERVICE_RULES);
+        logger.debug("Presented rules on switch {}: {}", switchId, presentedCookies);
+
+        return makeRulesResponse(expectedCookies, presentedCookies, switchId);
+    }
+
+    private Set<Long> readSegmentsCookies() {
+        String q = "MATCH (:switch)-[fs:flow_segment]->(sw:switch) WHERE sw.name=$switch_id RETURN fs";
+        StatementResult queryResults = session.run(q, ImmutableMap.of("switch_id", switchId.toString()));
+        return extractCookies(queryResults, "fs");
+    }
+
+    private Set<Long> readIngressCookies() {
+        String q = "MATCH (sw:switch)-[r:flow]->(:switch) WHERE sw.name=$switch_id RETURN r";
+        StatementResult queryResults = session.run(q, ImmutableMap.of("switch_id", switchId.toString()));
+        return extractCookies(queryResults, "r");
+    }
+
+    private Set<Long> extractCookies(StatementResult queryResults, String fieldName) {
+        return queryResults.list()
+                .stream()
+                .map(record -> Optional.ofNullable(record)
+                        .map(r -> r.get(fieldName))
+                        .map(r -> r.get("cookie"))
+                        .map(Value::asLong))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toSet());
+    }
+
+    private SyncRulesResponse makeRulesResponse(Set<Long> expectedCookies,
+                                                Set<Long> presentedCookies,
+                                                SwitchId switchId) {
+        Set<Long> missingRules = new HashSet<>(expectedCookies);
+        missingRules.removeAll(presentedCookies);
+        if (!missingRules.isEmpty()) {
+            logger.error("On switch {} the following rules are missed: {}", switchId, missingRules);
```

Comment:
Hex representation will be better.

Reply:
I think that logger call `SwitchId::toString()` and it will make hex representation.

Reply:
I am talk about cookies, not switch/datapath id. In decimal form cookie is unreadable.

Reply:
I see.